From a3dc81c69455f17e389d2c9a87b959586b0ff5ae Mon Sep 17 00:00:00 2001
From: Jared Baur <jaredbaur@fastmail.com>
Date: Sat, 18 Nov 2023 09:02:56 -0800
Subject: [PATCH 1/4] Add Kconfig VBOOT_SIGN option

This option allows for opting-out of signing during the coreboot build.
This can be useful when the build is to be performed in a reproducible
environment without access to any private signing keys.
---
 src/security/vboot/Kconfig      | 4 ++++
 src/security/vboot/Makefile.inc | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/security/vboot/Kconfig b/src/security/vboot/Kconfig
index 2d5b20f50d..32ce60825a 100644
--- a/src/security/vboot/Kconfig
+++ b/src/security/vboot/Kconfig
@@ -24,6 +24,10 @@ if VBOOT
 comment "Anti-Rollback Protection disabled because mocking secdata is enabled."
 	depends on VBOOT_MOCK_SECDATA
 
+config VBOOT_SIGN
+	bool "Perform signing during the coreboot build"
+	default n
+
 config VBOOT_SLOTS_RW_A
 	bool "Firmware RO + RW_A"
 	help
diff --git a/src/security/vboot/Makefile.inc b/src/security/vboot/Makefile.inc
index 1b3568a1ec..d341ccf24d 100644
--- a/src/security/vboot/Makefile.inc
+++ b/src/security/vboot/Makefile.inc
@@ -368,7 +368,7 @@ files_added:: $(FUTILITY)
 		$(obj)/coreboot.rom
 endif
 
-ifneq (,$(filter y,$(CONFIG_VBOOT_SLOTS_RW_A) $(CONFIG_VBOOT_SLOTS_RW_AB)))
+ifeq (yy,$(CONFIG_VBOOT_SIGN)$(filter y,$(CONFIG_VBOOT_SLOTS_RW_A) $(CONFIG_VBOOT_SLOTS_RW_AB)))
 files_added:: $(obj)/coreboot.rom $(FUTILITY) $(CBFSTOOL)
 	CBFSTOOL="$(CBFSTOOL)" \
 	$(FUTILITY) sign \
-- 
2.42.0

